runner/views/run_api_view.py

```diff
@@ -92,13 +93,25 @@ class OperatorViewSet(GenericAPIView):
     logger = logging.getLogger(__name__)
 
     def post(self, request):
-        request_ids = request.data['request_ids']
+        request_ids = request.data.get('request_ids', None)
+        run_ids = request.data.get('run_ids', None)
         pipeline_name = request.data['pipeline_name']
         pipeline = get_object_or_404(Pipeline, name=pipeline_name)
 
-        for request_id in request_ids:
-            logging.info("Submitting requestId %s to pipeline %s" % (request_id, pipeline_name))
-            create_jobs_from_request.delay(request_id, pipeline.operator_id)
+        if pipeline_name == "roslin-qc":
```

Comment:
Why aren't we handling roslin-qc operator the same way as any other operator? request_handlers shouldn't have any operator specific logic in them.

Reply:
because Roslin QC Operator requires that all `request_ids` and `run_ids` be passed, while other Operators only currently support a single `request_id` arg. Trying to refactor all other Operators to be able to use this would take a lot more work. Also, its less confusing when the logic for dictating the behavior of how the handling takes place is placed up front here near the endpoint/view, instead of buried way down inside some other module.